ReactAndroid/src/main/java/com/facebook/react/ReactActivityDelegate.java

```diff
@@ -79,17 +84,19 @@ public ReactInstanceManager getReactInstanceManager() {
   }
 
   protected void onCreate(Bundle savedInstanceState) {
+    boolean needsOverlayPermission = false;
     if (getReactNativeHost().getUseDeveloperSupport() && Build.VERSION.SDK_INT >= 23) {
       // Get permission to show redbox in dev builds.
       if (!Settings.canDrawOverlays(getContext())) {
-        Intent serviceIntent = new Intent(Settings.ACTION_MANAGE_OVERLAY_PERMISSION);
-        getContext().startActivity(serviceIntent);
+        needsOverlayPermission = true;
+        Intent serviceIntent = new Intent(Settings.ACTION_MANAGE_OVERLAY_PERMISSION, Uri.parse("package:" + getContext().getPackageName()));
         FLog.w(ReactConstants.TAG, REDBOX_PERMISSION_MESSAGE);
         Toast.makeText(getContext(), REDBOX_PERMISSION_MESSAGE, Toast.LENGTH_LONG).show();
```

Comment:
I think its still worth showing to the developer though so they understand what they need to do. Especially since we distribute to quality assurance it would help them understand the situation.